### PR TITLE
add fips compliant package builds

### PR DESCRIPTION
**Pull request by rajrohanyadav**
### Changes
* update goreleaser version
* add fips compliant package building
* update renovate config to update go version in Dockerfile builds

### Testing
[test s3 package for prerelease](http://nr-downloads-ohai-staging.s3-website-us-east-1.amazonaws.com/rohan-fips-test/)
`goreleaser` output with local testing:
```log
root@b6b613dee1a0:/app# bin/goreleaser release --config ./build/.goreleaser.yml --clean
  • cleaning distribution directory
  • loading environment variables
  • getting and validating git state
    • git state                                      commit=cb94086849408cb78901684dd57b1556f34339b3 branch=NR-287144-build-fips-compliant-packages current_tag=v2.1.0 previous_tag=v2.0.9 dirty=false
  ⨯ release failed after 0s                          error=git tag v2.1.0 was not made against commit cb94086849408cb78901684dd57b1556f34339b3
root@b6b613dee1a0:/app# bin/goreleaser release --config ./build/.goreleaser.yml --clean --snapshot
  • skipping announce, publish and validate...
  • cleaning distribution directory
  • loading environment variables
  • getting and validating git state
    • git state                                      commit=cb94086849408cb78901684dd57b1556f34339b3 branch=NR-287144-build-fips-compliant-packages current_tag=v2.1.0 previous_tag=v2.0.9 dirty=false
    • pipe skipped                                   reason=disabled during snapshot mode
  • parsing tag
  • setting defaults
    • pipe skipped                                   reason=release is disabled
  • snapshotting
    • building snapshot...                           version=2.1.0-SNAPSHOT-cb94086
  • ensuring distribution directory
  • setting up metadata
  • writing release metadata
  • loading go mod information
  • build prerequisites
  • building binaries
    • building                                       binary=dist/nri-nix_linux_arm_6/nri-docker
    • building                                       binary=dist/nri-nix_linux_amd64_v1/nri-docker
    • building                                       binary=dist/nri-nix_linux_arm64_v8.0/nri-docker
    • building                                       binary=dist/nri-nix_linux_386_sse2/nri-docker
    • building                                       binary=dist/nri-nix-fips_linux_arm64_v8.0/nri-docker
    • building                                       binary=dist/nri-nix-fips_linux_amd64_v1/nri-docker
    • took: 1m27s
  • archives
    • creating                                       archive=dist/nri-docker_linux_2.1.0-SNAPSHOT-cb94086_amd64_fips_dirty.tar.gz
    • creating                                       archive=dist/nri-docker_linux_2.1.0-SNAPSHOT-cb94086_arm64_dirty.tar.gz
    • creating                                       archive=dist/nri-docker_linux_2.1.0-SNAPSHOT-cb94086_arm_dirty.tar.gz
    • creating                                       archive=dist/nri-docker_linux_2.1.0-SNAPSHOT-cb94086_arm64_fips_dirty.tar.gz
    • creating                                       archive=dist/nri-docker_linux_2.1.0-SNAPSHOT-cb94086_amd64_dirty.tar.gz
    • creating                                       archive=dist/nri-docker_linux_2.1.0-SNAPSHOT-cb94086_386_dirty.tar.gz
  • calculating checksums
  • writing artifacts metadata
  • release succeeded after 1m30s
  • thanks for using goreleaser!
```
image

### CODE REVIEW

File: .github/workflows/on_prerelease.yaml [hunk 1]
```diff
@@ -9,13 +9,14 @@ on:
 
 jobs:
   pre-release:
-    uses: newrelic/coreint-automation/.github/workflows/reusable_pre_release.yaml@v3
+    uses: newrelic/coreint-automation/.github/workflows/reusable_pre_release.yaml@v4
     with:
```
Comment (alvarocabanas) [inline .github/workflows/on_prerelease.yaml:13]: Nice, and I could see the new version worked as expected.

File: .github/workflows/on_prerelease.yaml [hunk 2]
```diff
@@ -9,13 +9,14 @@ on:
...
       tag: ${{ github.event.release.tag_name }}
       integration: "docker"
       test_package: false
       run_test_windows: false
       run_build-win-packages: false
       publish_schema: custom
+      dest_prefix: "infrastructure_agent/"
```
Comment (alvarocabanas) [inline .github/workflows/on_prerelease.yaml:20]: You can leave it out and will take the default, isn't it?

File: build/.goreleaser.yml [hunk 3]
```diff
@@ -13,7 +16,25 @@ builds:
...
+        {{- if eq .Arch "arm64" -}}
+        CC=aarch64-linux-gnu-gcc
+        {{- end }}
+    goos:
+      - linux
+    goarch:
+      - amd64
+      - arm64
+    tags:
+      - fips
```
Comment (alvarocabanas) [inline build/.goreleaser.yml:37]: did you run `go tool nm nri-docker | grep FIPS` to ensure they are FIPS ?(They should be but just as an extra check)

**Review by alvarocabanas — COMMENTED** [review]
I left some comments

File: .github/workflows/on_prerelease.yaml (hunk 2 above)
Comment (rajrohanyadav) [reply .github/workflows/on_prerelease.yaml:20]: Yeah, I can leave it out. But left it as a documentation for what to change for testing. Probably I should add a comment. Let me add it.

File: .github/workflows/on_prerelease.yaml (hunk 1 above)
Comment (rajrohanyadav) [reply .github/workflows/on_prerelease.yaml:13]: Yeah, it worked. Thanks.

File: build/.goreleaser.yml (hunk 3 above)
Comment (rajrohanyadav) [reply build/.goreleaser.yml:37]: Adding the output here for ref.
```log
root@1b2ee7eb3e5d:/app/dist# go tool nm nri-nix-fips_linux_amd64_v1/nri-docker | grep FIPS
  462b80 t BORINGSSL_FIPS_abort
  45ea80 t FIPS_mode_set
  45ea90 t FIPS_read_counter
  4015e0 T _cgo_14d622f8b473_Cfunc__goboringcrypto_FIPS_mode
  45ea70 t _goboringcrypto_FIPS_mode
  743660 T crypto/internal/boring._Cfunc__goboringcrypto_FIPS_mode.abi0
  e23c58 D crypto/internal/boring._cgo_14d622f8b473_Cfunc__goboringcrypto_FIPS_mode
  740800 T crypto/internal/boring/sig.FIPSOnly.abi0
  e26bc0 D crypto/tls.defaultCipherSuitesFIPS
  e26b80 D crypto/tls.defaultCurvePreferencesFIPS
  e26ba0 D crypto/tls.defaultSupportedSignatureAlgorithmsFIPS
  e26b60 D crypto/tls.defaultSupportedVersionsFIPS
root@1b2ee7eb3e5d:/app/dist# go tool nm nri-nix_linux_amd64_v1/nri-docker | grep FIPS
reading nri-nix_linux_amd64_v1/nri-docker: no symbol section
reading nri-nix_linux_amd64_v1/nri-docker: no symbols
root@1b2ee7eb3e5d:/app/dist# 
```

Comment (alvarocabanas) [reply build/.goreleaser.yml:37]: Correct, thanks

**Review by alvarocabanas — APPROVED** [review]
LGTM

**Pull request merged**